stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/GroupChatInfoViewModel.kt

```diff
@@ -0,0 +1,132 @@
+                // Update members
+                _state.addSource(controller.members) { members ->
+                    val channelData = controller.channelData.value!!
+                    updateMembers(members, channelData.createdBy)
+                }
+
+                _state.addSource(controller.channelData) { channelData ->
+                    _state.value = _state.value?.copy(channelName = channelData.name)
+                }
+            }
+        }
+    }
+
+    fun onEvent(event: Event) {
+        when (event) {
+            is Event.NameChanged -> changeGroupName(event.name)
+            Event.MembersSeparatorClick -> _state.value = _state.value!!.copy(shouldExpandMembers = true)
+            is Event.MuteChannelClicked -> switchGroupMute(event.isEnabled)
+            is Event.ChannelMutesUpdated -> updateChannelMuteStatus(event.channelMutes)
+            Event.LeaveChannel -> leaveChannel()
+        }
+    }
+
+    private fun changeGroupName(name: String) {
+        viewModelScope.launch {
+            val result = channelClient.update(message = null, mapOf("name" to name)).await()
+            if (result.isError) {
+                // Handle error
+            }
+        }
+    }
+
+    private fun leaveChannel() {
+        viewModelScope.launch {
+            val result = chatDomain.useCases.hideChannel(cid, keepHistory = true).await()
+            if (result.isSuccess) {
+                _channelLeftState.value = true
+            } else {
+                // Handle error
+            }
+        }
+    }
+
+    private fun updateMembers(members: List<Member>, owner: User) {
+        val currentState = _state.value!!
+        val membersToShow = members.map {
+            ChatMember(it, isOwner = it.getUserId() == owner.id)
+        }
+        _state.value =
+            currentState.copy(
+                members = membersToShow,
+                shouldExpandMembers = currentState.shouldExpandMembers || members.size <= COLLAPSED_MEMBERS_COUNT,
+                membersToShowCount = members.size - COLLAPSED_MEMBERS_COUNT
+            )
+    }
+
+    private fun updateChannelMuteStatus(channelMutes: List<ChannelMute>) {
+        _state.value = _state.value!!.copy(channelMuted = channelMutes.any { it.channel.cid == cid })
+    }
+
+    private fun switchGroupMute(isEnabled: Boolean) {
+        viewModelScope.launch {
+            val result = if (isEnabled) {
+                channelClient.mute().await()
+            } else {
+                channelClient.unmute().await()
+            }
+            if (result.isError) {
+                // Handle error in a better way
+                _state.value = _state.value!!.copy(channelMuted = !isEnabled)
+            }
+        }
+    }
+
+    data class State(
+        val members: List<ChatMember> = emptyList(),
+        val channelName: String = "",
+        val channelMuted: Boolean = false,
+        val shouldExpandMembers: Boolean = false,
+        val membersToShowCount: Int = 0,
```

Comment:
As this contructor is only used once at the moment, I'd prefer removing the default values and providing them there instead. Makes for fewer unexpected usages of defaults later down the line.